packages/frontend-2/components/settings/Dialog.vue

```diff
@@ -0,0 +1,204 @@
+        <LayoutSidebarMenu>
+          <LayoutSidebarMenuGroup title="Account settings">
+            <template #title-icon>
+              <UserIcon class="h-5 w-5" />
+            </template>
+            <LayoutSidebarMenuGroupItem
+              v-for="(sidebarMenuItem, index) in menuItemConfig.user"
+              :key="`sidebarUserItem-${index}`"
+              :label="sidebarMenuItem.title"
+              :class="{
+                'bg-foundation-focus hover:!bg-foundation-focus':
+                  selectedMenuItem?.title === sidebarMenuItem.title
+              }"
+              @click="selectedMenuItem = sidebarMenuItem"
+            />
+          </LayoutSidebarMenuGroup>
+          <LayoutSidebarMenuGroup title="Server settings">
+            <template #title-icon>
+              <ServerStackIcon class="h-5 w-5" />
+            </template>
+            <LayoutSidebarMenuGroupItem
+              v-for="(sidebarMenuItem, index) in menuItemConfig.server"
+              :key="`sidebarServerItem-${index}`"
+              :label="sidebarMenuItem.title"
+              :class="{
+                'bg-foundation-focus hover:!bg-foundation-focus':
+                  selectedMenuItem?.title === sidebarMenuItem.title
+              }"
+              @click="selectedMenuItem = sidebarMenuItem"
+            />
+          </LayoutSidebarMenuGroup>
+        </LayoutSidebarMenu>
+      </LayoutSidebar>
+      <main
+        v-if="selectedMenuItem"
+        :class="[
+          'bg-foundation md:p-6 md:px-10 md:py-12 w-full md:bg-foundation',
+          !isMobile && 'simple-scrollbar overflow-y-auto flex-1'
+        ]"
+      >
+        <div class="pb-6">
+          <component :is="selectedMenuItem.component" />
+        </div>
+      </main>
+    </div>
+  </LayoutDialog>
+</template>
+
+<script setup lang="ts">
+import type { defineComponent } from 'vue'
+import type { LocationQueryValue } from 'vue-router'
+import SettingsUserProfile from './user/Profile.vue'
+import SettingsUserNotifications from './user/Notifications.vue'
+import SettingsUserDeveloper from './user/Developer.vue'
+import SettingsServerGeneral from './server/General.vue'
+import SettingsServerProjects from './server/Projects.vue'
+import SettingsServerActiveUsers from './server/ActiveUsers.vue'
+import SettingsServerPendingInvitations from './server/PendingInvitations.vue'
+import { useBreakpoints } from '@vueuse/core'
+import { TailwindBreakpoints } from '~~/lib/common/helpers/tailwind'
+import { UserIcon, ServerStackIcon } from '@heroicons/vue/24/outline'
+import { settingsQueries } from '~/lib/common/helpers/route'
+import {
+  LayoutSidebar,
+  LayoutSidebarMenu,
+  LayoutSidebarMenuGroup
+} from '@speckle/ui-components'
+
+type MenuItem = {
+  title: string
+  component: ReturnType<typeof defineComponent>
+}
+
+const emit = defineEmits<{
+  (e: 'update:open', val: boolean): void
+}>()
+
+const props = defineProps<{
+  open: boolean
+  targetMenuItem?: LocationQueryValue
```

Comment:
I changed to the `defineModel` and a `computed`, this was a nice suggestion made it a lot cleaner :) I kept the config inside the settings modal, as I feel only this component should actually care about the content of it, and the parent should just pass the query as a prop (and the modal can check wether it's a valid route or not, if not it can decide on the fallback)